src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,6 +219,41 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
             }
         }
 
+        private async Task LoadInitialConfiguration()
+        {
+            // Check if this is the first attempt to load the configuration for failed startup load as a part of refresh
+            if (MinCacheExpirationTime == null)
+            {
+                IEnumerable<KeyValueWatcher> watchers = _options.ChangeWatchers.Union(_options.MultiKeyWatchers);
+
+                if (!watchers.Any())
+                {
+                    // No registration was found for refresh
+                    return;
+                }
+
+                MinCacheExpirationTime = watchers.Min(w => w.CacheExpirationTime);
+            }
+
+            var timeElapsedSinceLastLoadAttempt = DateTimeOffset.UtcNow - LastRefreshAttemptTime;
+
+            if (timeElapsedSinceLastLoadAttempt < MinCacheExpirationTime || !RefreshAttemptSemaphore.Wait(0))
+            {
+                return;
+            }
+
+            LastRefreshAttemptTime = DateTimeOffset.UtcNow;

```

Comment:
>LastRefreshAttemptTime [](start = 12, length = 22)

looks like it's loadAttempt in the local variable. Load attempt makes more sense to me.

Reply:
Let's put a comment here to mark that this is the shared refresh attempt tracker.